pool/src/lib.rs

```diff
@@ -806,37 +1011,90 @@ impl LendingPool {
         env: &Env,
+        liquidator: Address,
         who: Address,
-        mb_who_balance: Option<(Address, i128)>,
-        mb_account_data: Option<AccountData>,
-        is_good: bool,
+        user_config: &mut UserConfiguration,
+        account_data: AccountData,
+        get_stoken: bool,
     ) -> Result<(), Error> {
-        let is_good_position =
-            Self::is_good_position(env, who, mb_who_balance, mb_account_data, is_good)?;
-        if !is_good_position {
-            return Err(Error::BadPosition);
+        let liquidation_data = account_data
+            .liquidation
+            .expect("pool: liquidation flag in calc_account_data");
+        let mut debt_with_penalty = liquidation_data.total_debt_with_penalty_in_xlm;
+        for collateral_to_receive in liquidation_data.collateral_to_receive {
+            if debt_with_penalty == 0 {
+                break;
+            }
+
+            let (reserve, s_token_balance, price_fixed, coll_coeff_fixed) =
+                collateral_to_receive.unwrap_optimized();
+            let price = FixedI128::from_inner(price_fixed);
+
+            let s_token = STokenClient::new(env, &reserve.s_token_address);
+            let underlying_asset = s_token.underlying_asset();
+            let coll_coeff = FixedI128::from_inner(coll_coeff_fixed);
+            let compounded_balance = coll_coeff
+                .mul_int(s_token_balance)
+                .ok_or(Error::LiquidateMathError)?;
+            let compounded_balance_in_xlm = price
+                .mul_int(compounded_balance)
+                .ok_or(Error::CalcAccountDataMathError)?;
+
+            let withdraw_amount_in_xlm = compounded_balance_in_xlm.min(debt_with_penalty);
+            // no overflow as withdraw_amount_in_xlm guaranteed less or equal than debt_to_cover
+            debt_with_penalty -= withdraw_amount_in_xlm;
+
+            let (s_token_amount, underlying_amount) =
+                if withdraw_amount_in_xlm != compounded_balance_in_xlm {
+                    let underlying_amount = price
+                        .recip_mul_int(withdraw_amount_in_xlm)
+                        .ok_or(Error::LiquidateMathError)?;
+                    let s_token_amount = coll_coeff
+                        .recip_mul_int(underlying_amount)
+                        .ok_or(Error::LiquidateMathError)?;
+                    (s_token_amount, underlying_amount)
+                } else {
+                    (s_token_balance, compounded_balance)
+                };
+
+            if get_stoken {
+                s_token.transfer_on_liquidation(&who, &liquidator, &s_token_amount);
+            } else {
+                s_token.burn(&who, &s_token_amount, &underlying_amount, &liquidator);
+            }
+
+            if s_token_balance == s_token_amount {
+                user_config.set_using_as_collateral(env, reserve.get_id(), false);
+                event::reserve_used_as_collateral_disabled(env, who.clone(), underlying_asset);
```

Comment:
Will this event be called twice within the liquidation?

First time within the `s_token.transfer_on_liquidation` (line 1071)
<img width="545" alt="image" src="https://github.com/eq-lab/Slender/assets/12034486/8bd9861f-9703-40cf-a6b2-4aaa49e75d54">


Second time, within the `do_liquidate`: 
<img width="609" alt="image" src="https://github.com/eq-lab/Slender/assets/12034486/ee312c83-1eb6-488e-84ca-13df115d6ec9">